### DISCUSSION TITLE
UDF invalid reference error

### DISCUSSION THREAD

**Discussion by rdlaitila** [geldata/gel#3383] [Q&A] [answered]
Wondering if I'm hitting a bug or unsupported EdgeQL language support for UDFs. For the given UDF:

```
module default {
	function test(foo: str, bar: str) -> str {
		using (
			with
				f := foo,
				b := bar,
				r := f ++ b,
			select r
		)
	}
}
```

When attempting to create the migrations via `edgedb migrations create` I get the following error:

```
edgedb error: InvalidReferenceError: schema item 'std::f' does not exist
```

Using edgedb-cli: 1.0.0-rc.6+4b9d831
Using edgedb-instance: 1.0-rc.4+1850eff

UDF documentation appears sparse for the early releases, wondering if this is just user error. Thanks!

**Comment by elprans** [comment, marked as answer]
It's a bug. Converting this into an issue.

**Reply by rdlaitila** [reply]
Thanks very much for the quick review! I'm currently exploring UDFs to see how much/little domain logic can live inside migration controlled database functions, akin to stored procs. Enjoying EdgeQL and the DDL very much will certainly be keeping an ear to the ground on this 👍
